code-reviews/readme.md

```diff
@@ -1,140 +1,160 @@
+4. Decide on a common standard for each language  
+    * Automate as much as possible (styling, etc.) to avoid the need for "Nit's" and allow the reviewer to focus more on functional aspects of the PR. 
+
+## First Design Pass
+
+1. Pull Request Overview 
+    1.  [ ] Does the PR description make sense? 
+    1.  [ ] Do all the changes logically fit in this PR, or are there unrelated changes? 
+    1.  [ ] If necessary, are the changes made reflected in updates to the README or other docs? If the changes affect how the user builds code especially. 
+
+1. User Facing Changes 
+    1.  [ ] If the code involves a user-facing change, is there a GIF/photo that explains the functionality? If not, it might be key to validate the PR to ensure the change does what is expected. 
+    1.  [ ] Ensure UI changes look good without unexpected behavior.  
+
+1. Design  
+    1.  [ ] Do the interactions of the various pieces of code in the PR make sense? 
+    1.  [ ] Does the code recognize and incorporate architectures and coding patterns? 
+
+## Code Quality Pass
+
+1. Complexity 
+    1.  [ ] Are functions too complex?  
+    1.  [ ] Should a function be broken into multiple functions? 
+    1.  [ ] If a method has greater than 3 arguments, potentially overly complex. 
+    1.  [ ] Single responsibility principle – function or class should do one ‘thing’  
+    1.  [ ] Can the code be understood easily by code readers?  
+    1.  [ ] Does the code add functionality that isn’t needed? Is it overly complex?  
+
+1. Naming/readability 
+    1.  [ ] Did the developer pick good names for functions, variables, etc? 
+
+1. Error Handling 
+    1.  [ ] Are errors handled gracefully and explicitly where necessary?  
+
+1. Functionality  
+    1.  [ ] Is there parallel programming in this PR that could cause race conditions? Carefully read through this logic. 
+    1.  [ ] Could the code be optimized? For example: are there more calls to the database than need be?  
+    1.  [ ] If you may not fully understand how the code could affect the system, ask for help. 
+    1.  [ ] Are there security flaws?  
+    1.  [ ] Does a variable name reveal any customer specific information?  
+    1.  [ ] Is PII and EUII treated correctly? And not logging any PII information. 
+
+1. Style  
+    1.  [ ] Are there extraneous comments? If the code isn’t clear enough to explain itself, then the code should be made simpler. Comments may be there to explain why some code exists.  
+    1.  [ ] Does the code adhere to the style guide/conventions that we have agreed upon? We use automated styling like black and prettier.  
+
+1. Tests 
+    1.  [ ] Tests should always be committed in the same PR as the code itself (‘I’ll add tests next’ is not acceptable) 
+    1.  [ ] Make sure tests are sensible and valid assumptions are made. 
+
+## Language Specific Guidance
+
+-   [Bash](recipes/Bash.md)
+-   [C#](recipes/CSharp.md)
+-   [Go](recipes/Go.md)
+-   [Java](recipes/Java.md)
```

Comment:
Java.md does not exist

Reply:
good catch!

---
In reply to: [379449240](https://github.com/microsoft/code-with-engineering-playbook/pull/148#discussion_r379449240) [](ancestors = 379449240)